app/User.php

```diff
@@ -244,6 +244,49 @@ public function getGreetingAndName()
         return $this->getGreeting() . ' ' . $this->getFullName();
     }
 
+    /**
+     * Get the user's record by Billet ID. For use in the retrieval of
+     * Space Lords and heads of branches, where a single user holds the
+     * specified billet.
+     *
+     * @param string $billet_id
+     *
+     * @return User|null
+     */
+    public static function getByBilletId(string $billet_id)
+    {
+        $billet = Billet::where('_id', $billet_id)->first();
```

Comment:
@cinnion While we're going to be calling this with curated billet ids, it's probably best if we still do some defensive programming here and check that we have a valid billet back

Reply:
Added more checks on whether we have a valid billet.